pkg/component/component.go

```diff
@@ -0,0 +1,229 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package component
+
+import (
+	"fmt"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+)
+
+// Unit is a single unit that needs to be running inside a component.
```

Comment:
Could we add more details about what is a unit in the context of the Elastic Agent, this include input, output.

Reply:
Will update.